src/unit_tests/analysisd/test_decoder_syscollector.c

```diff
@@ -1371,119 +2869,158 @@ void test_syscollector_dbsync_network_address_valid_msg_modified(void **state)
+            "Unable to send hotfixes information to Wazuh DB.");
+
+    int ret = DecodeSyscollector(lf, 0);
+
+    assert_int_not_equal(ret, -1);
+}
+
+void test_syscollector_hotfix_parse_result_not_ok (void **state)
+{
+    Eventinfo *lf = *state;
+    const char *result = "not_ok ";
     int sock = 1;
 
     expect_any(__wrap_wdbc_query_ex, *sock);
-    expect_string(__wrap_wdbc_query_ex, query, query);
+    expect_any(__wrap_wdbc_query_ex, query);
     expect_any(__wrap_wdbc_query_ex, len);
     will_return(__wrap_wdbc_query_ex, result);
     will_return(__wrap_wdbc_query_ex, 0);
+    expect_string(__wrap__mdebug1, formatted_msg, 
+            "Unable to send hotfixes information to Wazuh DB.");
 
     int ret = DecodeSyscollector(lf, &sock);
 
-    assert_int_not_equal(ret, 0);
+    assert_int_not_equal(ret, -1);
 }
-void test_syscollector_dbsync_hotfixes_valid_msg_inserted(void **state)
+
+void test_syscollector_netinfo_valid (void **state)
 {
     Eventinfo *lf = *state;
+    const char *queries[] = { "agent 001 netinfo save 100|2021/10/29 14:26:24|"
+                "86|87|88|89|99|90|91|92|93|94|95|96|97|98",
+            "agent 001 netproto save 100|86|0|0.0.0.2|0.0.0.3|10",
+            "agent 001 netaddr save 100|86|0|0.0.0.0|255.255.255.255|0.0.0.1",
+            "agent 001 netaddr save 100|86|0|0.0.1.0|255.255.255.254|0.0.1.1",
+            "agent 001 netproto save 100|86|1|0000:0000:0000:0:0000:0000:0000:0002|"
+                "0000:0000:0000:0:0000:0000:0000:0003|10",
+            "agent 001 netaddr save 100|86|1|0000:0000:0000:0:0000:0000:0000:0000|"
+                "0000::|0000:0000:0000:0:0000:0000:0000:0001", 
+            "agent 001 netaddr save 100|86|1|0000:0000:0000:0:0000:0000:0001:0000|"
+                "0001::|0000:0000:0000:0:0000:0000:0001:0001" };
 
-    const char *query = "agent 001 dbsync hotfixes INSERTED "
-        "{"
-            "\"scan_time\":\"2021/10/29 14:26:24\","
-            "\"hotfix\":\"KB123456\","
-            "\"checksum\":\"abcdef0123456789\""
-        "}";
     const char *result = "ok ";
     int sock = 1;
+    size_t count = sizeof(queries) / sizeof(*queries);
+
+    for (int i = 0; i < count; i++)
+    {
+        expect_any(__wrap_wdbc_query_ex, *sock);
+        expect_string(__wrap_wdbc_query_ex, query, queries[i]);
+        expect_any(__wrap_wdbc_query_ex, len);
+        will_return(__wrap_wdbc_query_ex, result);
+        will_return(__wrap_wdbc_query_ex, 0);
+    }
 
-    expect_any(__wrap_wdbc_query_ex, *sock);
-    expect_string(__wrap_wdbc_query_ex, query, query);
-    expect_any(__wrap_wdbc_query_ex, len);
-    will_return(__wrap_wdbc_query_ex, result);
-    will_return(__wrap_wdbc_query_ex, 0);
     int ret = DecodeSyscollector(lf, &sock);
 
     assert_int_not_equal(ret, 0);
 }
-void test_syscollector_dbsync_packages_valid_msg_inserted(void **state)
+
+static const char *valid_empty_queries[] = { "agent 001 netinfo save NULL|NULL|"
```

Comment:
In this approach, try to follow the general guideline of declaring variables before their use, which facilitates a better understanding of the code. However, from now on it can be subject to debate.